src/aliceVision/sfm/matchesFiltering.cpp

```diff
@@ -0,0 +1,87 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2016 AliceVision contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "matchesFiltering.hpp"
+
+namespace aliceVision {
+namespace sfm {
+
+void matchesGridFiltering(const aliceVision::feature::Regions& lRegions, const aliceVision::feature::Regions& rRegions,
```

Comment:
I guess u put it here because of the SfMData parameter.
Ideally, it would be better to have such function in `matching`.
Maybe, since SfMData is only needed to retrieve the size of the images, we can have this function with a different API in matching, like

```c++
void matchesGridFiltering(const aliceVision::feature::Regions& regionsLeft, std::pair<std::size_t, std::size_t>  imSizeLeft, , std::pair<std::size_t, std::size_t>  imSizeLeft, const aliceVision::feature::Regions& regionRight, aliceVision::matching::IndMatches& outMatches, size_t gridSize)
```
and here we can have this function that retrieves the necessary data and call the other one. What do you think?
It would be easier to write a unit test, for example, without passing by the SfMData just for the image size.

Reply:
Yes, you're right. I didn't pay attention to the content of the function. This is stupid.